components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMember.kt

```diff
@@ -9,14 +8,18 @@ import net.corda.membership.impl.registration.staticnetwork.StaticMemberTemplate
 import net.corda.membership.impl.registration.staticnetwork.StaticMemberTemplateExtension.Companion.STATIC_PLATFORM_VERSION
 import net.corda.membership.impl.registration.staticnetwork.StaticMemberTemplateExtension.Companion.STATIC_SERIAL
 import net.corda.membership.impl.registration.staticnetwork.StaticMemberTemplateExtension.Companion.STATIC_SOFTWARE_VERSION
+import net.corda.membership.lib.EndpointInfoFactory
 import net.corda.v5.membership.EndpointInfo
 import net.corda.utilities.time.UTCClock
 
 /**
  * Class which represents a static member. Static members are Map<String, Any> which we need
  * to map to Map<String, String>. This class provides access to properties always as strings.
  */
-class StaticMember(private val staticMemberData: Map<String, Any>) : Map<String, Any> by staticMemberData {
+class StaticMember(
+    private val staticMemberData: Map<String, Any>,
+    private val endpointInfoFactory: EndpointInfoFactory,
```

Comment:
A lambda could be better here so we don't have essentially a data class holding a reference to a service class.
```suggestion
    private val endpointInfoFactory: (String, String) -> EndpointInfoFactory,
```
(I might gotten the syntax wrong there, but something like that is what I mean)